Scripts/extras/pull_and_build_from_git.py

```diff
@@ -593,30 +649,51 @@ def build_and_install_cmake(self):
                         target_folder_path.mkdir(parents=True)
                     shutil.copy2(glob_result, str(target_folder_path.resolve()), follow_symlinks=False)
 
+    def create_custom_env(self):
+        custom_env = os.environ.copy()
+        custom_env['TARGET_INSTALL_ROOT'] = str(self.build_install_folder.resolve())
+        custom_env['PACKAGE_ROOT'] = str(self.package_install_root.resolve())
+        custom_env['TEMP_FOLDER'] = str(self.base_temp_folder.resolve())
+        if self.package_info.depends_on_packages:
+            package_folder_list = ""
+            for package_name, _, subfoldername in self.package_info.depends_on_packages:
+                if package_folder_list:
+                    package_folder_list += ";"
+                # note that it must be an abs path for cmake to accept it:
+                # The package declaration may specify a subfolder name but the path library will
+                # behave correctly if you slash operator with an empty (by ignoring it)
+                package_folder_list += str( (self.base_temp_folder / package_name / subfoldername).resolve().absolute())
+            package_folder_list = package_folder_list.replace('\\', '/')
+            custom_env['DOWNLOADED_PACKAGE_FOLDERS'] = package_folder_list
```

Comment:
minor

string.join can be used so you don't need an extra condition to add ';'

```suggestion
            package_folder_list = []
            for package_name, _, subfoldername in self.package_info.depends_on_packages:
                # note that it must be an abs path for cmake to accept it:
                # The package declaration may specify a subfolder name but the path library will
                # behave correctly if you slash operator with an empty (by ignoring it)
                package_folder_list += str( (self.base_temp_folder / package_name / subfoldername).resolve().absolute())
            package_folder_list = package_folder_list.replace('\\', '/')
            custom_env['DOWNLOADED_PACKAGE_FOLDERS'] = ';'.join(package_folder_list).replace('\\', '/')
```